ci_framework/roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -58,18 +58,13 @@
         mac_address: "{{ _lm_mac_address }}"
       ansible.builtin.import_tasks: reserve_ip.yml
 
+    # Note: This task will fail if the virt-net name is greater than 15 chars
     - name: Attach interface to the virtual machine.
-      vars:
-        _net_name: >
-          {{
-            (network.static_ip | bool) |
-            ternary(network.name, 'cifmw-net_' + network.name)
-          }}
       ansible.builtin.command:
         cmd: >-
           virsh -c qemu:///system
           attach-interface "{{ vm_name }}"
-          --source "{{ _net_name }}"
+          --source "{{ network.name | regex_replace('\n', '') }}"
```

Comment:
I have a stupid question, Does the virt net network name will contains new line? Like this `cifmw-net\nosp_trunk` then source will be return as `cifmw-netosp_trunk` based on above expression.

Reply:
I found the value of *source* was `cifmw-net_osp_trunk\n` in multiple runs. I couldn't figure out where the newline character was getting inserted. 

The only place we gather this information is from `json_query` and the output looked good upto the point of `command`.  Earlier we had

```
      vars:
        _net_name: >
          {{
            (network.static_ip | bool) |
            ternary(network.name, 'cifmw-net_' + network.name)
          }}
```

not sure if this had some influence. Hence the replace of carriage return.

Reply:
Thank you for explaining it. I think we can use `{{ network.name | trim }}` to remove the new line.

Reply:
Weird. I strongly believe something was happening in the `vars`. Now without trim or replace ... it is working.